library/src/main/java/com/google/android/cameraview/CameraView.java

```diff
@@ -428,30 +433,35 @@ public void remove(Callback callback) {
         public void onCameraOpened() {
             if (mRequestLayoutOnOpen) {
                 mRequestLayoutOnOpen = false;
-                requestLayout();
+                cameraView.requestLayout();
             }
             for (Callback callback : mCallbacks) {
-                callback.onCameraOpened(CameraView.this);
+                callback.onCameraOpened(cameraView);
             }
         }
 
         @Override
         public void onCameraClosed() {
             for (Callback callback : mCallbacks) {
-                callback.onCameraClosed(CameraView.this);
+                callback.onCameraClosed(cameraView);
             }
         }
 
         @Override
         public void onPictureTaken(byte[] data) {
             for (Callback callback : mCallbacks) {
-                callback.onPictureTaken(CameraView.this, data);
+                callback.onPictureTaken(cameraView, data);
             }
         }
 
         public void reserveRequestLayoutOnOpen() {
             mRequestLayoutOnOpen = true;
         }
+
+        private void cleanup() {
+            cameraView = null;
```

Comment:
Shouldn't there be a check as to whether cameraView.stop() has been called or not?